language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/eventsync/subscribers/PublishDiagnosticSubscriber.java

```diff
@@ -32,18 +33,22 @@
  */
 @JavaSPIService("org.ballerinalang.langserver.commons.eventsync.spi.EventSubscriber")
 public class PublishDiagnosticSubscriber implements EventSubscriber {
+
     public static final String NAME = "Publish diagnostic subscriber";
 
     @Override
     public EventKind eventKind() {
         return EventKind.PROJECT_UPDATE;
     }
-    
+
     @Override
     public void onEvent(ExtendedLanguageClient client, DocumentServiceContext context,
                         LanguageServerContext languageServerContext) {
-        DiagnosticsHelper diagnosticsHelper = DiagnosticsHelper.getInstance(languageServerContext);
-        diagnosticsHelper.compileAndSendDiagnostics(client, context);
+        LSClientCapabilities lsClientCapabilities = context.languageServercontext().get(LSClientCapabilities.class);
+        if (!lsClientCapabilities.getInitializationOptions().isEnableLSLightWeightMode()) {
```

Comment:
There's another place where we publish diagnostics, `BallerinaWorkspaceService.didChangeWatchedFiles()` method. Need to add this check there as well.